sphinxdocs/private/sphinx.bzl

```diff
@@ -275,10 +283,14 @@ def _run_sphinx(ctx, format, source_path, inputs, output_prefix):
     # Build in parallel, if possible
     # Don't add to run_args: parallel building breaks interactive debugging
     args.add("--jobs", "auto")
-    args.add("--fresh-env")  # Don't try to use cache files. Bazel can't make use of them.
-    run_args.append("--fresh-env")
-    args.add("--write-all")  # Write all files; don't try to detect "changed" files
-    run_args.append("--write-all")
+
+    if use_cache:
+        args.add("--doctree-dir", paths.join(output_dir.path, ".doctrees"))
```

Comment:
When you spin up a project with `sphinx-quickstart` and then run `make html` (or whatever) to build the docs, the doctree dir gets output as a sibling to the HTML output dir. But when you build from within Bazel, the doctrees dir does indeed get output as a child of the HTML output dir

Reply:
Per my other comment: I've refactored the code to put the doctree dir outside of the output dir. This is necessary because bazel will clear the output dir on every action re-run, thus losing the doctrees if they're in it.